<launch>
  <!-- launch file for state feedback -->
  <!-- Arguments -->

  <arg name="num_of_gmm_dist" default="3"/>

  <!-- campus route 1-->
  <!-- <arg name="orient_x" default="37.0"/>
  <arg name="orient_y" default="-97.0"/>
  <arg name="goal_x" default="55.0"/>
  <arg name="goal_y" default="-87.0"/> -->
  <arg name="orient_x" default="35.0"/>
  <arg name="orient_y" default="-96.0"/>
  <arg name="goal_x" default="38.0"/>
  <arg name="goal_y" default="-94.0"/>
  <!-- <arg name="goal_x" default="53.0"/>
  <arg name="goal_y" default="-84.0"/> -->

  <!-- campus route 2-->
  <!-- <arg name="orient_x" default="8.0"/>
  <arg name="orient_y" default="-45.5"/>
  <arg name="goal_x" default="29.5"/>
  <arg name="goal_y" default="-87"/> -->
  
  <arg name="gmm" default="true"/>
  <!-- <arg name="gmm" default="false"/> -->

  <arg name="no_gmm_speed" default="0.26"/>
  
  <!-- <param name="using_gmm_param" value="$(arg using_gmm)"/> -->
  <param name="orient_x" value="$(arg orient_x)"/>
  <param name="orient_y" value="$(arg orient_y)"/>
  <param name="goal_x" value="$(arg goal_x)"/>
  <param name="goal_y" value="$(arg goal_y)"/>
  <param name="num_of_gmm_dist" value="$(arg num_of_gmm_dist)"/>

  <param name="gmm" value="$(arg gmm)"/>

  <param name="no_gmm_speed" value="$(arg no_gmm_speed)"/>

  <!-- <node name="error_plot"
        pkg="rqt_plot"
        type="rqt_plot"
        args="/error/x" /> -->

  <!-- <node name="error_plot"
      pkg="rqt_plot"
      type="rqt_plot"
      args="/error/x /error/y" /> -->

  <!-- <node pkg="particle_to_gmm" type="gmm_visualizer.py" name="gmm_visualizer" output="screen">
  </node> -->

  <node pkg="particle_to_gmm" type="particle_to_gmm.py" name="particle_to_gmm" output="screen">
  </node>

  <node pkg="particle_to_gmm" type="gmm_controller_with_thread.py" name="gmm_controller" output="screen">
  </node> 
    <!-- Node of state feedback -->
  <!-- <node pkg="particle_to_gmm" type="state_feedback_line.py" name="state_feedback_line" output="screen">
  </node>  -->

  <node pkg="particle_to_gmm" type="ref_line" name="ref_line" output="screen">
  </node> 


</launch>
